.github/workflows/promote-canary-to-latest.yml

```diff
@@ -0,0 +1,34 @@
+# Workflow to manually promote the returntocorp/semgrep:canary docker image
+# to returntocorp/semgrep:latest that most customers are using
+
+name: Promote returntocorp/semgrep:canary to :latest
+
+on:
+  workflow_dispatch:
+    inputs:
+      confirmed:
+        description: "Are you sure you want to do this? This is a sensitive operation"
+        type: boolean
+        required: true
+        default: false
+
+jobs:
+  promote:
+    name: promote :canary to :latest
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - env:
+          confirmed: ${{ inputs.confirmed }}
+        run: |
+          canary_digest=$(docker buildx imagetools inspect --raw returntocorp/semgrep:canary | jq -r .config.digest)
```

Comment:
maybe a small comment here saying "
This is basically docker pull r2c/semgrep:canary; docker tag r2c/semgrep:canary r2c/semgrep:latest; docker push r2c/semgrep:latest
but do it instead in a more complex way to be multiarch friendly, so it handles our amd64 and arm64 docker images.
Note that if we were doing the simpler 3 commands above, we would just update the amd64 r2c/semgrep:latest docker image (because amd64 is the arch we use in this workflow). This would still work, but arm64 users would then pull this docker image and use emulation to run it (it would not pull the latest arm64 docker image!! so it would still be correct, but it would be slow).
For more infromation on multiarch docker build and buildx/imagetools and digest see URLHERE or good blog post to learn about it.

Reply:
Not to be pedantic but `docker buildx imagetools create` is pretty different from pull-tag-push, but I can add a comment explaining. Good call mentioning pull-tag-push as a fallback w/ caveats in case this doesn't work for some reason